### PR TITLE
[WIP] Feature: Improving Signing Data Documentation

### PR DESCRIPTION
**Pull request by BboyAkers**
Working on improving the Signing Data Documentation.

Clarification List:
- [x] The purpose of `EIP712Domain`
- [x] Create available `Primary Types` list
- [x] Domains and the current acceptable properties behind them.
- [x] Clarify use of `sigUtil.recoverTypedSignature_v4`
- [x] Clarify the use of `domain` and the difference between `EIP712Domain`

Reference:
https://eips.ethereum.org/EIPS/eip-712
https://github.com/danfinlay/js-eth-personal-sign-examples/blob/master/index.js#L322

### CODE REVIEW

File: docs/guide/signing-data.md [hunk 1]
```diff
@@ -57,13 +57,13 @@ If on-chain verifiability cost is a high priority for you, you might want to con
 
 ## Sign Typed Data v3
 
-The method `signTypedData_v3` represents a pretty recent version of the [EIP-712 spec](https://eips.ethereum.org/EIPS/eip-712), which at the time made it the most secure method for signing cheap-to-verify data on-chain that we had yet.
+The method `signTypedData_v3` currently represents the latest version of the [EIP-712 spec](https://eips.ethereum.org/EIPS/eip-712), making it the most secure method for signing cheap-to-verify data on-chain that we have yet.
```
Comment (danfinlay) [inline docs/guide/signing-data.md:60]: Oh is this highlighting that the `_v4` changes have not been merged into the EIP yet?

File: docs/guide/signing-data.md [hunk 2]
```diff
@@ -57,13 +57,13 @@ If on-chain verifiability cost is a high priority for you, you might want to con
...
 
-We intend to protect this namespace and keep it compatible going forwards.
+This does not mean it is perfect, and does not mean we will not eventually have a `v4`, but we do intend to protect this namespace and keep it compatible going forwards.
 
 We have a [great introductory blog post to this method here](https://medium.com/metamask/eip712-is-coming-what-to-expect-and-how-to-use-it-bb92fd1a7a26).
 
-Hopefully soon we will also have good examples for parsing method input into structs for verification on-chain (great contribution opportunity!)
+Hopefully soon we will also have good examples for paOh rsing method input into structs for verification on-chain (great contribution opportunity!)
```
Comment (danfinlay) [inline docs/guide/signing-data.md:66]:
```suggestion
Hopefully soon we will also have good examples for parsing method input into structs for verification on-chain (great contribution opportunity!)
```

**Review by danfinlay — CHANGES_REQUESTED** [review]
Just a question and a typo. Otherwise, thanks so much for putting this together, this is so long needed!

File: docs/guide/signing-data.md (hunk 1 above)
Comment (BboyAkers) [reply docs/guide/signing-data.md:60]: I think so.

Comment (danfinlay) [reply docs/guide/signing-data.md:60]: Ok I'll open an issue for that...

File: docs/guide/signing-data.md [hunk 3]
```diff
@@ -57,9 +57,9 @@ If on-chain verifiability cost is a high priority for you, you might want to con
 
 ## Sign Typed Data v3
 
-The method `signTypedData_v3` represents a pretty recent version of the [EIP-712 spec](https://eips.ethereum.org/EIPS/eip-712), which at the time made it the most secure method for signing cheap-to-verify data on-chain that we had yet.
+The method `signTypedData_v3` currently represents the latest version of the [EIP-712 spec](https://eips.ethereum.org/EIPS/eip-712), making it the most secure method for signing cheap-to-verify data on-chain that we have yet.
 
-We intend to protect this namespace and keep it compatible going forwards.
+This does not mean it is perfect, and does not mean we will not eventually have a `v4`, but we do intend to protect this namespace and keep it compatible going forwards.
```
Comment (danfinlay) [inline docs/guide/signing-data.md:62]: I just want to include at least a nod to v4.
```suggestion
This does not mean it is perfect, and we do already have a `v4` in prototype stage (which supports recursive structs and arrays), but we do intend to protect this namespace and keep it compatible going forwards.
```

**Review by danfinlay — APPROVED** [review]

**Pull request merged**